### DISCUSSION TITLE
ActionPlugin getting the data submitted by POST request

### DISCUSSION THREAD

**Discussion by Zontex** [inventree/InvenTree#2722] [Q&A] [answered]
I'm trying to make a custom action plugin that recieve data and perform action. to begin with, id like a very simlple example of a plugin that recieve a string such as item_name and respond with the same item name it recieved. in the samples I can only find the API to perform action and give response but not how to use the data from the plugin to perform an action, how to do that? thanks!

**Comment by matmair** [comment, marked as answer]
@Zontex  The docs are your friend here: https://inventree.readthedocs.io/en/latest/extend/plugins/action/ example code is shipped in the [codebase](https://github.com/inventree/InvenTree/blob/master/InvenTree/plugin/builtin/action/simpleactionplugin.py).

The current user can be accessed as `self.user`, the provided data as `self.data` as seen in the [mixin definition](https://github.com/inventree/InvenTree/blob/8a8fd1b4ba8174dfb0ab96a9ec98cb074337ed87/InvenTree/plugin/builtin/action/mixins.py#L33-L68).

**Reply by matmair** [reply]
Please mark as an answer if everything is answered

**Reply by Zontex** [reply]
I went through the docs before posting and I used the exact same script to try execute a demo, I feel the documentation is not very documented and dififcult to get started. 

while it explains the function of each API such as what action api is responsible for, there are not many exmples or explaination how to use it, now I see the mixin definition which makes sense and explains, I'll try to make a demo and perhaps share my plugin so other can refer to it as well when making their own, thanks!

**Reply by matmair** [reply]
@zontex we are always happy to review PRs that improve things.
The docs for each mixin just describe the function, the general plugin docs give more examples and we ship sample plugins with the code. Maxbe that structure is not clear enough from the architecture overview.
